Shell/NewCNF.cpp

```diff
@@ -404,7 +404,7 @@ TermList NewCNF::findITEs(TermList ts, Stack<unsigned> &variables, Stack<Formula
 }
 
 bool NewCNF::shouldInlineITE(unsigned iteCounter) {
-  return iteCounter < _iteInliningThreshold;
+  return _iteInliningThreshold == 0 || iteCounter < _iteInliningThreshold;
```

Comment:
I don't know. Could you expand your reasoning? I see:

`_iteInliningThreshold((unsigned)ceil(log2(namingThreshold)))`

and am wondering what cpp computes for log2(0) that it does not crash. Could you first figure that out?

Then naming is never 1, and from 2 on, we get something slowly growing along with naming threshold. You feel there should be a special value saying "always inline"? (I am not saying there shoudn't...) Or would that be especially useful for your case here?

Reply:
The point here would be that induction expects the clauses from `NewCNF` in a specific form, and naming ruins it. How about using `_forInduction` to always inline? That would not change the normal behaviour since the flag is only true when we clausify induction formulas.

Reply:
Sorry for the delay, I wanted to see where the ITE comes from and I couldn't find a related benchmark. Actually, in `NewCNF::toClauses` it tries to "inline" FOOL true and false when `gc->foolBindings` is non-empty. I think since there can an arbitrary number of such bindings, we cannot set the threshold to a great enough value to always inline, so adding a special case when e.g. `_forInduction` is true could still be a better solution here.

Reply:
I think I am getting more confused about what's going on. Maybe a short call on skype could be a good way to clarify?